integration-tests/src/tests/client/sharding_upgrade.rs

```diff
@@ -650,15 +652,6 @@ fn setup_genesis(
     genesis.config.avg_hidden_validator_seats_per_shard =
         epoch_config.avg_hidden_validator_seats_per_shard;
 
-    // TODO(resharding)
-    // In the current simple test setup each validator may track different
-    // shards. Unfortunately something is broken in how state sync is triggered
-    // from this test. The config below forces both validators to track all
-    // shards so that we can avoid that problem for now. It would be nice to
-    // set it up properly and test both state sync and resharding - once the
-    // integration is fully supported.
-    genesis.config.minimum_validators_per_shard = 2;
```

Comment:
Is `.track_all_shards()` a fix for this issue?

Reply:
yes